src/main/controllers/TypeOfClaimController.ts

```diff
@@ -43,26 +44,29 @@ export default class TypeOfClaimController {
   }
 
   public post = (req: AppRequest, res: Response): void => {
-    const redirectUrl = conditionalRedirect(req, this.form.getFormFields(), [TypesOfClaim.BREACH_OF_CONTRACT])
-      ? AuthUrls.LOGIN
-      : LegacyUrls.ET1_BASE;
+    // const redirectUrl = conditionalRedirect(req, this.form.getFormFields(), [TypesOfClaim.BREACH_OF_CONTRACT])
```

Comment:
Why is this commented out code left in?

Reply:
For this I have had to update the existing conditionalRedirect function due to recent updates in latest master, in short, as debugged locally it used to store an array within array for the form field in the req.body, but after the update in master yesterday that is no longer the case (happy to demonstrate it locally if required). Now with my changes, I have tested that when Breach of Contract is selected it would redirect to the legacy page, otherwise it would redirect to our login page.

Reply:
Long story short, I've had to update the conditionalRedirect function, as it is no longer working properly after the recent changes in the parser of how form field data are parsed. In my understanding, prior that changes the form field would be stored as an array regardless of whether single or multi fields are selected, whereas after that changes, it would only store the values as an array if multiple fields are selected (please find attached screenshots).
I have updated the conditionalRedirect function so that now it will handle both single and multi fields data and return the correct boolean.

![before_single](https://user-images.githubusercontent.com/101253854/162453393-13bdd00e-7649-4254-bf6d-035f2b862408.PNG)
![current_multi](https://user-images.githubusercontent.com/101253854/162453396-22c4b85f-037e-4a2d-b9b2-57345870c714.PNG)
![current_single](https://user-images.githubusercontent.com/101253854/162453398-3e7507ff-61bd-48eb-8ade-e2028c197c81.PNG)